src/main/java/com/solvd/micro9/tickets/service/impl/EsTicketCommandHandlerImpl.java

```diff
@@ -0,0 +1,148 @@
+                .build();
+        return esEventRepository.findByEntityId(command.getTicket().getEventId())
+                .collectList()
+                .map(esEventsList -> {
+                    if (esEventsList.isEmpty()) {
+                        throw new ResourceDoesNotExistException(
+                                "Event [id=" + command.getTicket().getEventId() + "] does not exist"
+                        );
+                    } else {
+                        return esEventsList;
+                    }
+                })
+                .zipWith(esTicketRepository.save(event))
+                .map(Tuple2::getT2)
+                .doOnSuccess(esTicket -> producer.send("New event", esTicket));
+    }
+
+    @Override
+    public void apply(SetTicketsUserIdToNullByUserIdCommand command) {
+        List<EsTicket> esTicketList = new ArrayList<>();
+        final boolean[] isStreamCompleted = {false};
+        esTicketRepository.findAll()
+                .filter(esTicket -> {
+                    Ticket ticket = new Gson().fromJson(esTicket.getPayload(), Ticket.class);
+                    return esTicket.getType().equals(EsType.TICKET_CREATED)
+                            && command.getUserId().equals(ticket.getUserId());
+                })
+                .doOnNext(esTicket -> {
+                    String template = "{\"userId\":\"%s\"}";
+                    EsTicket event = EsTicket.builder()
+                            .type(EsType.TICKET_USER_DELETED)
+                            .time(LocalDateTime.now())
+                            .createdBy(command.getCommandBy())
+                            .entityId(esTicket.getEntityId())
+                            .payload(String.format(template, command.getUserId()))
+                            .status(EsStatus.SUBMITTED)
+                            .build();
+                    esTicketList.add(event);
+                })
+                .doOnComplete(() -> isStreamCompleted[0] = true)
+                .subscribeOn(Schedulers.boundedElastic())
+                .subscribe();
+
+        while (!isStreamCompleted[0]) {
+        } //TODO is there a better way to do?
+
+        Flux.fromIterable(esTicketList)
+                .flatMap(esTicket -> esTicketRepository.save(esTicket)
+                        .doOnSuccess(event -> {
+                            TicketCache ticketCache = new TicketCache(
+                                    event.getId(),
+                                    command.getUserId()
+                            );
+                            cache.put(RedisConfig.CACHE_KEY, event.getId(), ticketCache)
+                                    .subscribeOn(Schedulers.boundedElastic())
+                                    .subscribe();
+                            producer.send("New event", event);
+                        })
+                )
+                .subscribeOn(Schedulers.boundedElastic())
+                .subscribe();
```

Comment:
you subscribes in this mehtod twicely, that's not good practice
[first](
https://github.com/eyakauleva/ticket-microservice/pull/19/files#diff-d8d3b07fee98e5459b29350f259b5ca997b336a07aeb16e3c0fdab0befaa1e29R107) and now